### DISCUSSION TITLE
Need help using javascript filter

### DISCUSSION THREAD

**Discussion by shalomabitan** [zendesk/maxwell#2083] [General]
Hi! New to this package and can use some help. I'm using Kinesis as the Producer for our maxwell image. However, we're using Datadog and want to dump the output of the row for logging purposes. here's the question. in `process_row()` if we do this 

```js
function process_row(row) {
  if (row) {
    var eventType = row.database + "." + row.table + "." + row.type;
    var rowData = JSON.stringify({
      eventType: eventType,
      data: row.data,
      old_data: row.old_data
    });

    logger.info(rowData);
  }
return row;
}
```

The do not show `row.data` or `row.old_data`. In fact for some reason they come back undefined. Now, if we simply use `logger.info(row.data)` it's all good. 

Any suggestions on how to get this to display properly? 

Thanks in advance!

**Comment by osheroff** [comment]
moved to discussions...

**Comment by osheroff** [comment]
this is utterly annoying and I spent awhile trying to bludgeon the nashorn API into doing something smarter than this.  I failed.  this will work tho:

```
function process_row(row) {
   if (row) {
      var data = {}
      for each (var p in row.data.keySet()) {
         data[p] = row.data[p]
      }

      var eventType = row.database + "." + row.table + "." + row.type;
      var rowData = JSON.stringify({
         eventType: eventType,
         data: data,
         old_data: row.old_data
      });

      logger.info(rowData);
   }
   return row;
}
```

**Reply by osheroff** [reply]
(obv do the same for `row.old_data`)

**Comment by osheroff** [comment]
the problem here seems to be that JSON.stringify doesn't like working with non-javascript-native objects.  I'm sure there's a solution here in `WrappedRowMap` but right now it eludes me.

**Comment by shalomabitan** [comment]
@osheroff Thank you very much!! As i get more familiar I hope to be able to contribute 🙏🏻
